pkg/virt-operator/resource/generate/components/prometheus.go

```diff
@@ -282,31 +281,30 @@ func NewPrometheusRuleSpec(ns string, workloadUpdatesEnabled bool) *v1.Prometheu
 					{
 						Record: "vec_by_virt_operators_all_client_rest_requests_in_last_hour",
 						Expr: intstr.FromString(
-							fmt.Sprintf("sum by (pod) (sum_over_time(rest_client_requests_total{pod=~'virt-operator-.*', namespace='%s'}[60m]))", ns),
+							generateValueDifferenceInHourQuery(generateAllPodRequestsQuery("virt-operator", ns, "")),
 						),
 					},
 					{
 						Record: "vec_by_virt_operators_failed_client_rest_requests_in_last_hour",
 						Expr: intstr.FromString(
-							fmt.Sprintf("sum by (pod) (sum_over_time(rest_client_requests_total{pod=~'virt-operator-.*', namespace='%s', code=~'(4|5)[0-9][0-9]'}[60m]))", ns),
+							generateValueDifferenceInHourQuery(generateAllPodRequestsQuery("virt-operator", ns, "(4|5)[0-9][0-9]")),
 						),
 					},
 					{
 						Record: "vec_by_virt_operators_all_client_rest_requests_in_last_5m",
 						Expr: intstr.FromString(
-							fmt.Sprintf("sum by (pod) (sum_over_time(rest_client_requests_total{pod=~'virt-operator-.*', namespace='%s'}[5m]))", ns),
+							generateValueDifferenceInFiveMinutesQuery(generateAllPodRequestsQuery("virt-operator", ns, "")),
 						),
 					},
 					{
 						Record: "vec_by_virt_operators_failed_client_rest_requests_in_last_5m",
 						Expr: intstr.FromString(
-							fmt.Sprintf("sum by (pod) (sum_over_time(rest_client_requests_total{pod=~'virt-operator-.*', namespace='%s', code=~'(4|5)[0-9][0-9]'}[5m]))", ns),
+							generateValueDifferenceInFiveMinutesQuery(generateAllPodRequestsQuery("virt-operator", ns, "(4|5)[0-9][0-9]")),
 						),
 					},
 					{
 						Alert: "VirtOperatorRESTErrorsHigh",
 						Expr:  intstr.FromString("(vec_by_virt_operators_failed_client_rest_requests_in_last_hour / vec_by_virt_operators_all_client_rest_requests_in_last_hour) >= 0.05"),
-						For:   "5m",
```

Comment:
@Barakmor1 IIUC here you are changing the functionality of `VirtOperatorRESTErrorsHigh`. What you wanted to achieve by removing the `For: "5m"` is an immediate alert, right? 
Isn't that required to be defined in another alert? because IIUC the aim of `VirtOperatorRESTErrorsHigh` is to alert about a constant failure of REST calls over time. For example having REST errors of 10-20 rate per sample for the whole 5 minutes.

Reply:
if i understood correctly i think that firing is a state and that when we add For: "5m" to the alert
it means that Prometheus will fire an alert only if the condition (in the Expr's value) is true for 5m
so it would be a west of time to wait another 5 minutes when the error rate is high for at least one hour.